### PR TITLE
Followup fix ASTRenameQuery::clone 

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

followup  #61013

### CODE REVIEW

File: src/Parsers/ASTRenameQuery.h [hunk 1]
```diff
@@ -94,7 +94,24 @@ class ASTRenameQuery : public ASTQueryWithOutput, public ASTQueryWithOnCluster
     ASTPtr clone() const override
     {
         auto res = std::make_shared<ASTRenameQuery>(*this);
-        res->cloneChildren();
```
Comment (vdimir) [inline src/Parsers/ASTRenameQuery.h:97]: `cloneChildren` do not update elements, so we have old ASTs there, we should update them also in `clone`

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 87c9652d0dda38724a46610a62357e3927d8308a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61254/87c9652d0dda38724a46610a62357e3927d8308a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Parsers/ASTRenameQuery.h [hunk 2]
```diff
@@ -94,7 +94,24 @@ class ASTRenameQuery : public ASTQueryWithOutput, public ASTQueryWithOnCluster
...
+        res->children.clear();
+
+        auto clone_child = [&res](ASTPtr & node)
+        {
+            if (node)
+            {
+                node = node->clone();
```
Comment (Algunenano) [inline src/Parsers/ASTRenameQuery.h:103]: Maybe it's me, but readability of this isn't great (will `elem.from.database` be updated? Yes, but it's not obvious). Not sure how to make it better without duplicating code though, so feel free to ignore it.

**Review by Algunenano — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by vdimir** [comment]
> Stateless tests (release) — fail: 1, passed: 6195, skipped: 4 [Details](https://s3.amazonaws.com/clickhouse-test-reports/61254/87c9652d0dda38724a46610a62357e3927d8308a/stateless_tests__release_.html)

https://github.com/ClickHouse/ClickHouse/issues/60480

> Stress test (tsan) — Possible deadlock on shutdown (see gdb.log) [Details](https://s3.amazonaws.com/clickhouse-test-reports/61254/87c9652d0dda38724a46610a62357e3927d8308a/stress_test__tsan_.html)

https://github.com/ClickHouse/ClickHouse/issues/49156

**Comment by vdimir** [comment]
> AST fuzzer (ubsan) — Logical error: 'Expected single dictionary argument for function.'. [Details](https://s3.amazonaws.com/clickhouse-test-reports/61254/c7e8ca789f8bffa4451765bd51882014ed1768c3/ast_fuzzer__ubsan_.html)

:-|

> Integration tests (asan, analyzer) [4/6] — fail: 1, passed: 500 [Details](https://s3.amazonaws.com/clickhouse-test-reports/61254/c7e8ca789f8bffa4451765bd51882014ed1768c3/integration_tests__asan__analyzer__[4_6].html)

https://github.com/ClickHouse/ClickHouse/issues/55168

> Stateless tests (debug) [3/5] — fail: 2, passed: 1262, skipped: 23 [Details](https://s3.amazonaws.com/clickhouse-test-reports/61254/c7e8ca789f8bffa4451765bd51882014ed1768c3/stateless_tests__debug__[3_5].html)

`01508_partition_pruning_long`, `02125_lz4_compression_bug`

https://github.com/ClickHouse/ClickHouse/issues/60854#issuecomment-2003648049

**Pull request merged**